libnetwork/internal/filelease/filelease.go

```diff
@@ -0,0 +1,129 @@
+// Package filelease implements "best effort" file locking, for Linux hosts, by
+// wrapping fcntl(2)'s F_SETLEASE.
+//
+// A file that is bind mounted into a container (for example "/etc/hosts",
+// "/etc/resolv.conf") can't be updated by moving a new version into place. So,
+// when updating it, it's possible for consumers inside the container to see a
+// truncated or incomplete version of the file. If a write lease can be obtained
+// for one of these files, processes attempting to open the file in the container
+// will block until the lease is dropped.
+//
+// If the file already has open descriptors, a lease cannot be obtained. OpenFile
+// can be configured to retry, and to return an open file descriptor even if the
+// lease cannot be acquired.
+package filelease
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/containerd/log"
+	"github.com/docker/docker/libnetwork/types"
+	"github.com/pkg/errors"
+)
+
+// Use OpenFile to open a file, attempt to acquire a write lease, and
+// construct a FileLease.
+type FileLease struct {
+	*os.File
+	leaseErr error
+}
+
+// Opts describes options for NewFileLeaser.
+type Opts struct {
+	flags    int           // flag for os.OpenFile()
+	mode     os.FileMode   // perm for os.OpenFile()
+	attempts int           // Max attempts to acquire lease.
+	interval time.Duration // Interval between attempts to acquire lease.
+}
+
+// DefaultOpts returns a default set of options that can be modified if necessary.
```

Comment:
```suggestion
// defaultOpts returns a default set of options that can be modified if necessary.
```

Reply:
Done - thanks ... and added missing comments to the related "WithXX()" functions that were late additions.